controllers/openstackdataplanenode_controller.go

```diff
@@ -146,6 +133,10 @@ func (r *OpenStackDataPlaneNodeReconciler) Reconcile(ctx context.Context, req ct
 			// and recalculate it based on the state of the rest of the conditions
 			instance.Status.Conditions.Set(instance.Status.Conditions.Mirror(condition.ReadyCondition))
 		}
+
+		// Ensure conditions are always sorted by type
+		instance.Status.Conditions.Sort()
```

Comment:
If this is really needed then I think we have a bug in lib-common. I think all the condition modifying function in lib-common ensure that the list is sorted.

Reply:
Thanks, I've removed these lines. They are indeed sorted in the call to Set().